core/src/main/java/kafka/server/SharePartitionManager.java

```diff
@@ -631,4 +651,112 @@ public ShareFetchPartitionData(FetchParams fetchParams, String groupId, String m
+
+        public static final String SHARE_ACK_SENSOR = "share-acknowledgement-sensor";
+        public static final String SHARE_ACK_RATE_PER_MINUTE = "share-acknowledgement-rate-per-minute";
+        public static final String SHARE_ACK_COUNT = "share-acknowledgement-count";
+
+        public static final String RECORD_ACK_SENSOR = "record-acknowledgement";
+        public static final String RECORD_ACK_RATE_PER_MINUTE = "record-acknowledgement-rate-per-minute";
+        public static final String RECORD_ACK_COUNT = "record-acknowledgement-count";
+        public static final String ACK_TYPE = "ack-type";
+
+        public static final String PARTITION_LOAD_TIME_SENSOR = "partition-load-time-sensor";
+        public static final String PARTITION_LOAD_TIME_AVG = "partition-load-time-avg";
+        public static final String PARTITION_LOAD_TIME_MAX = "partition-load-time-max";
+
+
+        private static final Map<Byte, String> RECORD_ACKS_MAP = Utils.mkMap(
+            Utils.mkEntry((byte) 1, "accept"),
+            Utils.mkEntry((byte) 2, "release"),
+            Utils.mkEntry((byte) 3, "reject")
+        );
+
+        private final Metrics metrics;
+        private final Time time;
+        private final Sensor shareAcknowledgementSensor;
+        private final Map<Byte, Sensor> recordAcksSensorMap = new HashMap<>();
+        private final Sensor partitionLoadTimeSensor;
+
+        public ShareGroupMetrics(Metrics metrics, Time time) {
+            this.metrics = metrics;
+            this.time = time;
+
+            shareAcknowledgementSensor = metrics.sensor(SHARE_ACK_SENSOR);
+            shareAcknowledgementSensor.add(metrics.metricName(
+                    SHARE_ACK_COUNT,
+                    METRICS_GROUP_NAME,
+                    "The total number of offsets acknowledged for share groups.",
+                    PROTOCOL, GROUP_STATE),
+                new CumulativeSum());
+            shareAcknowledgementSensor.add(metrics.metricName(
+                    SHARE_ACK_RATE_PER_MINUTE,
+                    METRICS_GROUP_NAME,
+                    "The total number of offsets acknowledged for share groups per minute.",
+                    PROTOCOL, GROUP_STATE),
+                new Rate(TimeUnit.MINUTES));
+
+            for (Map.Entry<Byte, String> entry : RECORD_ACKS_MAP.entrySet()) {
+                recordAcksSensorMap.put(entry.getKey(), metrics.sensor(String.format("%s-%s-sensor", RECORD_ACK_SENSOR, entry.getValue())));
+                recordAcksSensorMap.get(entry.getKey())
+                    .add(metrics.metricName(
+                            RECORD_ACK_RATE_PER_MINUTE,
+                            METRICS_GROUP_NAME,
+                            "The number of records acknowledged per acknowledgement type per minute.",
+                            ACK_TYPE, entry.getValue(),
+                            PROTOCOL, GROUP_STATE),
+                        new Rate(TimeUnit.MINUTES));
```

Comment:
Why to have `new Rate(TimeUnit.MINUTES)` intead of `new Rate()` which will be in seconds?

Reply:
because I had kept the rate calculations per minute (as suggested by the name of the metric)